guides/source/error_reporting.md

```diff
@@ -8,53 +8,48 @@ This guide introduces ways to manage exceptions that occur in Ruby on Rails appl
 
-Error Reporter
+Error Reporting
 ------------------------
 
-The [error reporter](https://api.rubyonrails.org/classes/ActiveSupport/ErrorReporter.html) collects exceptions occur in Ruby on Rails applications and reports them to registered subscribers.
-
-The goals are to
-
-1. Automatically capture and report any unhandled exception from a controller or a job.
+The Rails [error reporter](https://api.rubyonrails.org/classes/ActiveSupport/ErrorReporter.html) provides a standard way to collect exceptions that occur in your application and report them to your preferred service or location.
 
-2. Replace such manual rescue in applications and libraries
+The error reporter aims to replace boilerplate error-handling code like this:
 
-```rb
+```ruby
 begin
   do_something
 rescue SomethingIsBroken => error
   MyErrorReportingService.notify(error)
 end
 ```
 
-with
+with a consistent interface:
 
-```rb
+```ruby
 Rails.error.handle(SomethingIsBroken) do
   do_something
 end
 ```
 
-This approach provides several benefits:
+Rails wraps all executions (such as HTTP requests, jobs, and `rails runner` invocations) in the error reporter, so any unhandled errors raised in your app will automatically be reported to your error-reporting service via their subscribers.
+
+This means that third-party error-reporting libraries no longer need to insert a Rack middleware or do any monkey-patching to capture unhandled exceptions. Libraries that use ActiveSupport can also use this to non-intrusively report warnings that would previously have been lost in logs.
+
+Using the Rails' error reporter is not required. All other means of capturing errors still work.
 
-* Application or error reporting libraries don't need to insert a Rack middleware to capture unhandled exceptions from requests anymore.
-* To ActiveSupport-aware libraries, this can be used to report errors to the host application.
-* It decouples application code from error reporting libraries.
-* It reduces boilerplate code for handling exceptions.
-* Error reporting libraries will need less monkey-patches and be less intrusive to applications.
+### Subscribing to the Reporter
 
-### Subscribe To The Reporter
+To use the error reporter, you need a _subscriber_. A subscriber is any object with a `report` method. When an error occurs in your application or is manually reported, the Rails error reporter will call this method with the error object and some options.
 
-An error subscriber is expected to have a `report` method that takes an exception object and a few options.
-For example:
+Some error-reporting libraries, such as [Sentry's](https://github.com/getsentry/sentry-ruby/blob/master/sentry-rails/lib/sentry/rails/error_subscriber.rb) and [Honeybadger's](https://docs.honeybadger.io/lib/ruby/integration-guides/rails-exception-tracking/), automatically register a subscriber for you. Consult your provider's documentation for more details.
```

Comment:
Should we change that sentry URL to point to a specific commit? I'm worried that the file may be moved or whatever which would break the link. 

So that would be https://github.com/getsentry/sentry-ruby/blob/e18ce4b6dcce2ebd37778c1e96164684a1e9ebfc/sentry-rails/lib/sentry/rails/error_subscriber.rb

Reply:
👍